### PR TITLE
Allow sending nested JSON string to parseStreamMessage

### PR DESCRIPTION
**Pull request by shanshanzhu**
Fix issue in https://github.com/uber/xviz/issues/335

**Why**
when we send the data to `parseStreamMessage` in the shape of JSON or Uint8Array, and when web worker is used, the data is mutated when we are using webworker.
The goal is to save a snapshot of data that can't be mutated (string) and has minimal perf impact.

**What is in the PR**
1. Move JSON.parse to decode and decode is called recursively. Now the parser supports JSON in this shape:

```
{
  type: 'snapshot',
  updates: '//xviz string that can be sent to JSON.parse'
}
```
2. Added Bench tests for additional data shapes as described in 2.

**No Significant Performance impact**

_Bench results with this fix branch_
We can see that the text-encoder has minimal impact on performance.
screen shot 2019-02-20 at 12 54 56 pm

_Bench on master_
screen shot 2019-02-20 at 12 35 48 pm

### CODE REVIEW

File: modules/parser/src/parsers/parse-stream-data-message.js [hunk 1]
```diff
@@ -79,13 +91,7 @@ export function parseStreamDataMessage(message, onResult, onError, opts) {
   }
 
   try {
-    let data;
-    if (typeof message === 'string') {
-      data = JSON.parse(message);
```
Comment (Pessimistress) [inline modules/parser/src/parsers/parse-stream-data-message.js]: What is the motivation for this change - Is there a user case where message is string but not JSON string?

Comment (shanshanzhu) [reply modules/parser/src/parsers/parse-stream-data-message.js]: Because when we send the data in the shape of JSON or Uint8Array to parse function, and when web worker is used, the data is mutated.

The goal is to save a snapshot of data that can't be mutated and with minimal perf impact.
